File: 완전탐색/프로그래머스_카펫.js
function solution(brown, yellow) {
    var width; // 가로
    var height; // 세로
    
    var total = brown+yellow;
    
    for(height=1; height<total; height++){
        if(total%height==0){
            width = total/height;
        }
        
        if((width+height-2) == (brown/2)){
            return [width,height];
         }
    }
}  
 
var brown = 24;  
var yellow = 24;   

console.log(solution(brown,yellow)); 

File: 완전탐색/프로그래머스_카펫.java
public class 프로그래머스_카펫 {
    public static void main(String[] args) {
        int brown = 10;
        int yellow = 2;

        System.out.println(solution(brown, yellow));
    }

    private static int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int h = 0;
        int w = 0;

        int total = brown+yellow;

        for(int i = 1; i<total; i++){
            if(total%i==0){
                h = i;
                w = total/h;
            }

            if((h+w-2)==brown/2){
                if(h>w){
                    answer[0] = h;
                    answer[1] = w;
                }else{
                    answer[0] = w;
                    answer[1] = h;
                }
                break;
            }
        }

        return answer;
    }
}
